src/main/webapp/app/file-upload-submission/file-upload-submission.component.ts

```diff
@@ -85,14 +93,30 @@ export class FileUploadSubmissionComponent implements OnInit, ComponentCanDeacti
                         this.numberOfAllowedComplaints = allowedComplaints;
                     });
                 }
+                if (this.submission.submitted) {
+                    this.setSubmittedFile();
+                }
 
-                if (data.submissions && data.submissions.length > 0) {
-                    this.submission = data.submissions[0] as FileUploadSubmission;
-                    if (this.submission && this.submission.submitted) {
-                        this.setSubmittedFile();
+                if (submission.result) {
+                    if (this.submission.submitted && submission.result.completionDate) {
+                        this.isTimeOfComplaintValid = this.resultService.isTimeOfComplaintValid(submission.result, this.fileUploadExercise);
+                        this.fileUploadAssessmentService.getAssessment(this.submission.id).subscribe((assessmentResult: Result) => {
+                            this.result = assessmentResult;
+                        });
+                        this.complaintService
```

Comment:
This nested structure makes it hard to understand the code (https://en.wikipedia.org/wiki/Pyramid_of_doom_(programming)).

With RxJS you can easily chain multiple steps on the same level:

```
service.get(abc).pipe(
  map(sum),
  tap(informClient),
  switchMap(() => nowGetOtherThing(def))
).subscribe();

Reply:
I simplified if statement by setting `this.result` beforehand and not making a huge `if else`. This also fixes the bug with complaint not loaded properly. Not sure if any RxJS changes are needed here now.